scripts/lib/dependencies.ts

```diff
@@ -67,9 +67,71 @@ export interface Dependency {
   getLatestVersion(): Promise<string | AlpineLimaISOVersion>
 }
 
+/**
+ * Types that implement UnreleasedChangeMonitor can tell you whether
+ * there have been any changes in their repository since their last release.
+ */
+export interface UnreleasedChangeMonitor {
+  hasUnreleasedChanges(): Promise<HasUnreleasedChangesResult>
+}
+
+export type HasUnreleasedChangesResult = {latestReleaseTag: string, hasUnreleasedChanges: boolean};
+
+type GithubRelease = Awaited<ReturnType<Octokit['rest']['repos']['listReleases']>>['data'][0];
+
+async function getLatestPublishedRelease(githubOwner: string, githubRepo: string): Promise<GithubRelease> {
+  const response = await getOctokit().rest.repos.listReleases({ owner: githubOwner, repo: githubRepo });
+
+  for (const release of response.data) {
+    if (release.published_at !== null) {
+      return release;
+    }
+  }
+  throw new Error(`Did not find a published release for ${ githubOwner }/${ githubRepo }`);
+}
+
+/**
+ * Tells the caller whether the given github repo has any
+ * changes that have not been released.
+ */
+export async function hasUnreleasedChanges(githubOwner: string, githubRepo: string): Promise<HasUnreleasedChangesResult> {
+  const latestRelease = await getLatestPublishedRelease(githubOwner, githubRepo);
+
+  // Get the date of the commit that the release's tag points to.
+  // We can't use the publish date of the release, because that
+  // omits commits that were made after the commit that was tagged
+  // for the release, but before the actual release.
+  const result = await getOctokit().rest.repos.getCommit({
+    owner: githubOwner, repo: githubRepo, ref: latestRelease.tag_name,
+  });
+  const dateOfTaggedCommit = result.data.commit.committer?.date;
+
+  if (!dateOfTaggedCommit) {
+    throw new Error(`Unexpected falsy value of dateOfTaggedCommit: ${ dateOfTaggedCommit }`);
+  }
+
+  // need to increment the date by 1s to avoid getting the tagged commit
+  const parsedDate = new Date(dateOfTaggedCommit);
+
+  parsedDate.setUTCSeconds(parsedDate.getUTCSeconds() + 1);
+  const incrementedDateOfTaggedCommit = parsedDate.toISOString();
+
+  const response = await getOctokit().rest.repos.listCommits({
+    owner: githubOwner, repo: githubRepo, since: incrementedDateOfTaggedCommit,
```

Comment:
So this is listing all commits from the default branch that's been committed starting from one second after the tag date. I think it would slightly cleaner to include the tag commit, and just check that `commits.length > 1` instead? Not entirely sure, though…

Reply:
Yeah I think that would be fine. Let's do that.